examples/src/main/java/com/mapbox/navigation/examples/core/BaseRouterActivityJava.java

```diff
@@ -0,0 +1,273 @@
+        "2019_04_13-00_00_11");
+    File fileTiles = new File(file, "tiles");
+    MapboxOnboardRouterConfig config = new MapboxOnboardRouterConfig(
+        fileTiles.getAbsolutePath(),
+        null,
+        null,
+        null,
+        null // working with pre-fetched tiles only
+    );
+
+    return new MapboxOnboardRouter(MapboxNativeNavigatorImpl.INSTANCE, config);
+  }
+
+  public static Router setupHybridRouter(Context context) {
+    return new MapboxHybridRouter(
+        setupOnboardRouter(),
+        setupOffboardRouter(context),
+        new NetworkStatusService(context)
+    );
+  }
+
+  private Router router;
+  private MapboxMap mapboxMap;
+
+  private DirectionsRoute route;
+  private NavigationMapRoute navigationMapRoute;
+  private Point origin;
+  private Point destination;
+  private Point waypoint;
+  private SymbolManager symbolManager;
+
+  @BindView(R.id.mapView)
+  MapView mapView;
+
+  abstract Router setupRouter();
+
+  @Override
+  protected void onCreate(@Nullable Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+    setContentView(R.layout.activity_mock_navigation);
+    ButterKnife.bind(this);
+
+    router = setupRouter();
+
+    mapView.onCreate(savedInstanceState);
+    mapView.getMapAsync(this);
+  }
+
+  @OnClick(R.id.newLocationFab)
+  public void onNewLocationClick() {
+    newOrigin();
+  }
+
+  private void newOrigin() {
+    if (mapboxMap != null) {
+      clearMap();
+      LatLng latLng = Utils.getRandomLatLng(new double[] { -77.1825, 38.7825, -76.9790, 39.0157 });
+      origin = Point.fromLngLat(latLng.getLongitude(), latLng.getLatitude());
+      mapboxMap.moveCamera(CameraUpdateFactory.newLatLngZoom(latLng, 12));
+    }
+  }
+
+  @Override
+  public void onMapReady(@NonNull MapboxMap mapboxMap) {
+    this.mapboxMap = mapboxMap;
+    this.mapboxMap.addOnMapClickListener(this);
+    mapboxMap.setStyle(Style.MAPBOX_STREETS, style -> {
+      Drawable image = ContextCompat.getDrawable(this, R.drawable.mapbox_marker_icon_default);
+      style.addImage(MARKER_ROUTE, image);
+      navigationMapRoute = new NavigationMapRoute(mapView, mapboxMap);
+      symbolManager = new SymbolManager(mapView, mapboxMap, style);
+      Snackbar.make(findViewById(R.id.container), "Tap map to place waypoint", Snackbar.LENGTH_LONG)
+          .show();
+      newOrigin();
+    });
+  }
+
+  private void clearMap() {
+    symbolManager.deleteAll();
+    route = null;
+    destination = null;
+    waypoint = null;
+    navigationMapRoute.updateRouteVisibilityTo(false);
+    navigationMapRoute.updateRouteArrowVisibilityTo(false);
+  }
+
+  private void findRoute() {
+    if (origin != null && destination != null) {
+      router.cancel();
+      if (TurfMeasurement.distance(origin, destination, TurfConstants.UNIT_METERS) > 50) {
+        RouteOptions.Builder optionsBuilder = applyDefaultParams(RouteOptions.builder())
+            .accessToken(Utils.getMapboxAccessToken(this))
+            .annotations(
```

Comment:
We missed `profile` 👀 https://github.com/mapbox/mapbox-navigation-android/blob/422e0a05bb65f16f7fd78bbeaf6087f2f40fa57a/examples/src/main/java/com/mapbox/navigation/examples/core/OffboardRouterActivityJava.java#L145

Reply:
OnboardActivity used a default `PROFILE_DRIVING` , and 
OffboardActivity used `PROFILE_DRIVING_TRAFFIC`.
Can I use the same for both cases? If yes, should I use `PROFILE_DRIVING` or `PROFILE_DRIVING_TRAFFIC`?

Reply:
_Onboard_ router is going to eventually support traffic so I believe we should be good using `PROFILE_DRIVING_TRAFFIC` for both. Currently _Onboard_ is `PROFILE_DRIVING` by default because at the moment is not implemented but if I'm not mistaken it's ignored by the backend. Is that right @danpaz? We can easily test too and see what's the behavior.

Reply:
ok, will use `PROFILE_DRIVING_TRAFFIC ` in both cases for now